### DISCUSSION TITLE
read_wkt to 'any' geometry

### DISCUSSION THREAD

**Discussion by JD31** [boostorg/geometry#1240] [Q&A] [answered]
I want to use read_wkt but I don't know in advance the type of the geometry.
Is read_wkt compatible with a geometry "super" type, or a variant ?
Or do I need to try/catch all options ?
Thank you.

**Comment by vissarion** [comment, marked as answer]
If I understand your question correctly you can use GEOMETRYCOLLECTION (is implemented in boost geometry but not documented yet). 

Here is an example that uses a variant of a point or polygon.

```cpp
    using Point = boost::geometry::model::d2::point_xy<double>;
    using Polygon = boost::geometry::model::polygon<Point>;
    using GeometryCollection = boost::geometry::model::geometry_collection
        <
            boost::variant<Polygon, Point>
        >;
    GeometryCollection gc;

    boost::geometry::read_wkt
        ("GEOMETRYCOLLECTION(POINT(0 0))", gc);

    std::cout << boost::geometry::wkt(gc) << std::endl;

    auto point = gc[0];
    std::cout << boost::geometry::wkt(point) << std::endl;
```
